### PR TITLE
Void Observations

### PR DESCRIPTION
**Pull request by G1DR4**
- WORK IN PROGRESS -

### CODE REVIEW

**Comment by dancunningham** [comment]
Some suggestions based on an initial look at this:
1. It doesn’t appear to be working. I opened patient ABC/ABC/135 and voided the final temp observation for 9th Oct (36.1) but in the grid it still shows 36.1 and when I tap to open the dialog it is still there [OK, it isn't hooked up yet]
2. I think the design needs some attention, especially for the “see every obs for a concept” and “see every obs on one day” views. The dialogs take up a lot of vertical space and aren’t that clear to read when there are lots and lots of observations. It’s fine for now, as those aren’t features required in this sprint. we can prioritise doing more work on those in a future sprint
3. I think the Void Observations toggle OFF/ON should just be a button
4. Can you modify the design to be laid out like this:

_​_Temperature on 9 Oct 2015_​_
———————————
10:10 - 36.7
11:10 - 36.8
21:10 - 36.1

a) Concept name and date only need to be shown once
b) Date format (at the moment it’s displaying as MM/DD/YYYY for me)
c) Order them the other way around, so the oldest is at the top and the newest is at the bottom

5: In the Void Observations dialog, it’s unclear whether the checkbox refers to the item above it or below it. This should be laid out like this:

_​_Void Observations_​_
———————————
_​_Temperature on 9 Oct 2015_​_

[  ] 10:10 - 36.7
[  ] 11:10 - 36.8
[  ] 21:10 - 36.1

[Cancel] [Void selected]

File: app/src/main/assets/chart.html [hunk 1]
```diff
@@ -77,35 +77,25 @@
...
-                    {% set output = summaryValue | format_value(row.item.captionFormat) %}
-                  {% endif %}
-                  {{output}}
-                {% endif %}
-              {% endif %}
-            </td>
+              class="{{column.start == nowColumnStart ? 'now' : ''}} {{class}}"
+              style="{{style}}"
+              onclick="{% if points is not empty%}
+                       {% if (row.item.type).string != 'text_icon' %}
```
Comment (capnfabs) [inline app/src/main/assets/chart.html:83]: There's a lot of duplicated logic in these conditionals. Could you clean that up with a variable?

File: app/src/main/assets/chart.html [hunk 2]
```diff
@@ -77,35 +77,25 @@
...
+                       {% endif %}">
+            {% if points is not empty %}
+            {% if row.item.type == 'text_icon' %}
+            <div>&#x1f4dd;</div>
+            {% else %}
+            {% set output = summaryValue | format_value(row.item.format) %}
+            {% if output is empty %}
+            {% set output = summaryValue | format_value(row.item.captionFormat) %}
+            {% endif %}
+            {{output}}
```
Comment (capnfabs) [inline app/src/main/assets/chart.html]: Please clean up the formatting here, indentation is especially important for these nested `if` statements.

File: app/src/main/assets/chart.html (hunk 1 above)
Comment (G1DR4) [inline app/src/main/assets/chart.html:83]: ok

File: app/src/main/assets/chart.html (hunk 2 above)
Comment (G1DR4) [inline app/src/main/assets/chart.html]: i'll do a clean up before merge request ;)

**Comment by llvasconcellos** [comment]
It's difficult to event test this branch because it does not sync anymore.
Gives this exception:

``` java
12-21 18:33:05.676   32651-5167/org.projectbuendia.client.dev E/buendia/SyncAdapter﹕ Error during sync
    net.sqlcipher.database.SQLiteConstraintException: error code 19: NOT NULL constraint failed: sync_tokens.sync_token
            at net.sqlcipher.database.SQLiteStatement.native_execute(Native Method)
            at net.sqlcipher.database.SQLiteStatement.execute(SQLiteStatement.java:58)
            at net.sqlcipher.database.SQLiteDatabase.insertWithOnConflict(SQLiteDatabase.java:1652)
            at net.sqlcipher.database.SQLiteDatabase.replaceOrThrow(SQLiteDatabase.java:1572)
            at org.projectbuendia.client.providers.GroupProviderDelegate.insert(GroupProviderDelegate.java:57)
            at org.projectbuendia.client.providers.GroupProviderDelegate.insert(GroupProviderDelegate.java:29)
            at org.projectbuendia.client.providers.DelegatingProvider.insert(DelegatingProvider.java:55)
            at org.projectbuendia.client.providers.BuendiaProvider.insert(BuendiaProvider.java:18)
            at android.content.ContentProvider$Transport.insert(ContentProvider.java:239)
            at android.content.ContentProviderClient.insert(ContentProviderClient.java:204)
            at org.projectbuendia.client.sync.SyncAdapter.storeSyncToken(SyncAdapter.java:590)
            at org.projectbuendia.client.sync.SyncAdapter.updatePatients(SyncAdapter.java:425)
            at org.projectbuendia.client.sync.SyncAdapter.onPerformSync(SyncAdapter.java:244)
            at android.content.AbstractThreadedSyncAdapter$SyncThread.run(AbstractThreadedSyncAdapter.java:259)
```

**Comment by llvasconcellos** [comment]
At least the patients synced.

**Comment by llvasconcellos** [comment]
OK. After rebase sync is restored, the observations are voided but the app throws an fatal exception:

``` java
12-21 19:18:36.417  23933-23933/org.projectbuendia.client.dev E/AndroidRuntime﹕ FATAL EXCEPTION: main
    Process: org.projectbuendia.client.dev, PID: 23933
    java.lang.NullPointerException: Attempt to invoke interface method 'void com.android.volley.Response$Listener.onResponse(java.lang.Object)' on a null object reference
            at com.android.volley.toolbox.JsonRequest.deliverResponse(JsonRequest.java:65)
            at com.android.volley.ExecutorDelivery$ResponseDeliveryRunnable.run(ExecutorDelivery.java:99)
            at android.os.Handler.handleCallback(Handler.java:739)
            at android.os.Handler.dispatchMessage(Handler.java:95)
            at android.os.Looper.loop(Looper.java:211)
            at android.app.ActivityThread.main(ActivityThread.java:5373)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1020)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:815)
```

**Comment by llvasconcellos** [comment]
Everything is working now. Good to go.

**Pull request merged**